sys/net/network_layer/ipv6/addr/ipv6_addr.c

```diff
@@ -52,25 +52,15 @@ uint8_t ipv6_addr_match_prefix(const ipv6_addr_t *a, const ipv6_addr_t *b)
     }
 
     for (int i = 0; i < 16; i++) {
-        /* if bytes are equal add 8 */
-        if (a->u8[i] == b->u8[i]) {
-            prefix_len += 8;
+        uint8_t xor = (a->u8[i] ^ b->u8[i]);
+        if (xor) {
+            /* if bytes aren't equal count matching leading bits */
+            prefix_len += __builtin_clz(xor) - 8 * (sizeof(int) - 1);
```

Comment:
Since `xor` is `uint8_t` the `sizeof(int)` is non-obvious. I think the argument to `__builtin_clz()` is `unsinged`, so `sizeof(unsigned)` is IMO easier to read. Still, a comment would be helpful.